packages/backend/src/apps/postman/actions/send-transactional-email/index.ts

```diff
@@ -96,35 +114,57 @@ const action: IRawAction = {
         replyTo: result.data.replyTo,
         senderName: result.data.senderName,
         attachments: attachmentFiles,
+      },
+    )
+
+    /**
+     * Patch the status of the recipients that were sent in the previous execution
+     */
+    if (isPartialRetry) {
+      const prevDataOut = prevDataOutParseResult.data
+      const updatedStatus = prevDataOut.status.map((oldStatus, i) => {
+        const correspondingRecipient = prevDataOut.recipient[i]
+        if (recipientsToSend.includes(correspondingRecipient)) {
+          return dataOut.status[
+            recipientsToSend.indexOf(correspondingRecipient)
+          ]
+        }
+        return oldStatus
       })
```

Comment:
bikeshed to reduce to O(n):

```suggestion
      const latestRecipientStatus = new Map(
        // dataOut.recipient is equal or subset of recipientsToSend.
        zip(dataOut.recipient, dataOut.status),
      )
      const updatedStatus = prevDataOut.status.map((oldStatus, i) => {
        const correspondingRecipient = prevDataOut.recipient[i]
        return latestRecipientStatus.get(correspondingRecipient) ?? oldStatus
      })
```